modules/jetstream-maxtext-deployment/main.tf

```diff
@@ -43,13 +44,20 @@ resource "kubernetes_manifest" "jetstream-service" {
 }
 
 resource "kubernetes_manifest" "jetstream-podmonitoring" {
-  count = var.maxengine_deployment_settings.metrics_port != null ? 1 : 0
+  count = var.maxengine_deployment_settings.metrics_port != null && var.maxengine_deployment_settings.metrics_scrape_interval != null ? 1 : 0
```

Comment:
This && behavior mixed with the metrics_scrape_interval not having a default set means that this variable is required to be set to deploy the jetstream workload podmonitoring object.

Is this the desired behavior, or should there be a default metrics scrape interval so this will deploy when a metrics port is set?

Reply:
The idea is that if the user doesn't want to scrape metics at all they leave the scrape interval field empty, but good call we should make the comments more descriptive. Likely going to follow this PR up at some point by reorganizing this modules parameters

Reply:
sgtm, comment updates sounds like a good enough solution for now to clarify the usage :)